ios/RCTTWVideoModule.m

```diff
@@ -77,7 +79,44 @@ @implementation RCTTWVideoModule
 
 RCT_EXPORT_MODULE();
 
+- (id)init {
+    self = [super init];
+    if (self) {
+        __weak RCTTWVideoModule *weakSelf = self;
+        self.statsReceivedCallback = ^(NSArray<TVIStatsReport *> * _Nonnull statsReports) {
+            NSMutableDictionary *eventBody = [[NSMutableDictionary alloc] initWithCapacity:10];
+            for (TVIStatsReport *statsReport in statsReports) {
+                NSMutableArray *audioTrackStats = [[NSMutableArray alloc] initWithCapacity:10];
+                NSMutableArray *videoTrackStats = [[NSMutableArray alloc] initWithCapacity:10];
+                NSMutableArray *localAudioTrackStats = [[NSMutableArray alloc] initWithCapacity:10];
+                NSMutableArray *localVideoTrackStats = [[NSMutableArray alloc] initWithCapacity:10];
+                for (TVIRemoteAudioTrackStats *stats in statsReport.remoteAudioTrackStats) {
+                    [audioTrackStats addObject:[weakSelf convertRemoteAudioTrackStats:stats]];
+                }
+                for (TVIRemoteVideoTrackStats *stats in statsReport.remoteVideoTrackStats) {
+                    [videoTrackStats addObject:[weakSelf convertRemoteVideoTrackStats:stats]];
+                }
+                for (TVILocalAudioTrackStats *stats in statsReport.localAudioTrackStats) {
+                    [localAudioTrackStats addObject:[weakSelf convertLocalAudioTrackStats:stats]];
+                }
+                for (TVILocalVideoTrackStats *stats in statsReport.localVideoTrackStats) {
+                    [localVideoTrackStats addObject:[weakSelf convertLocalVideoTrackStats:stats]];
+                }
+                eventBody[statsReport.peerConnectionId] = @{
+                    @"remoteAudioTrackStats": audioTrackStats,
+                    @"remoteVideoTrackStats": videoTrackStats,
+                    @"localAudioTrackStats": localAudioTrackStats,
+                    @"localVideoTrackStats": localVideoTrackStats
+                };
+            }
+            [weakSelf sendEventCheckingListenerWithName:statsReceived body:eventBody];
```

Comment:
Same here: lots of data processing is going on here. What do you thing about subscribing to specific stats?

Reply:
I thought about it, but it will require some additional work and the performance impact is not clear right now. I will probably make this optimization in a different PR.